config/run.tf

```diff
@@ -55,7 +55,8 @@ resource "google_cloud_run_service" "gcs_to_bq_service" {
 
         resources {
           limits = {
-            memory = "2G"
+            memory = "8G"
+            cpu = "2000m"
```

Comment:
@vanshkumar So initially I set this to 4GB and it was fine for population, but still ran out of memory for I think household income. The issue is that since this is adding breakdown metadata columns to every dataset that has a race breakdown for convenience on the frontend, it adds a lot of extra raw data. Apparently you can't set memory greater than 4GB without increasing CPU, which defaults to 1000m so this doubles it. Not sure if this is an issue.

There are a few alternatives:
* Move the addition of the metadata columns to SQL in the aggregation step, and just use the race_category_id col in pandas. This would need to be done for every dataset.
* Hard-code the race_category_id => (race, race_and_ethnicity, race_includes_hispanic) mapping on the frontend and process the datasets on the frontend before using them.
* Chunk the writing to BQ so it doesn't write all rows at once
* A larger refactor which I emailed you about - I think this would be really good to do post-launch, but is definitely too much work to do before launch.

Given that the pipeline isn't run very often, my take is to keep the memory/cpu increase and then look into the larger refactor post-launch. Happy to hear yours/others' thoughts or discuss further though!

Reply:
Thanks for listing the alternatives, I'm not sure what the cost will be of increasing mem to 8GB and doubling cpu, but as the cloud run service is up at all times, I imagine there will be some cost - I'll try to figure out how much that might be.

IMO at least until launch, it's OK to up this to 4GB and leave the cpu unchanged; household income is not a variable that we are planning to include by launch anyways. We can punt for now and a do a proper data pipeline refactor post launch

Reply:
I'll need to retest all the datasets and make sure HHI is the only one that didn't work with 4GB

Reply:
> Move the addition of the metadata columns to SQL in the aggregation step, and just use the race_category_id col in pandas. This would need to be done for every dataset.

Why would this need to be for every dataset? If the metadata columns are added in the aggregation step then jobs that need to use that data could read from the final table instead of the intermediate one output by gcs_to_bq.

Reply:
I'm not sure I follow what you're saying Jennie - could you elaborate?

Vansh - I updated to 4GB. All the pipelines run except for household income. I'll file a github issue for that.